src/API/Google/AdsCampaign.php

```diff
@@ -76,24 +78,30 @@ public function __construct( GoogleAdsClient $client, AdsCampaignBudget $budget
 	}
 
 	/**
-	 * Returns a list of campaigns
+	 * Returns a list of campaigns with targeted locations retrieved from campaign criterion.
 	 *
 	 * @return array
 	 * @throws ExceptionWithResponseData When an ApiException is caught.
 	 */
 	public function get_campaigns(): array {
 		try {
-			$return  = [];
-			$results = ( new AdsCampaignQuery() )
+			$campaign_results = ( new AdsCampaignQuery() )
 				->set_client( $this->client, $this->options->get_ads_id() )
 				->where( 'campaign.status', 'REMOVED', '!=' )
 				->get_results();
 
-			foreach ( $results->iterateAllElements() as $row ) {
-				$return[] = $this->convert_campaign( $row );
+			$campaign_criterion_results = ( new AdsCampaignCriterionQuery() )
+				->set_client( $this->client, $this->options->get_ads_id() )
+				->where( 'campaign.status', 'REMOVED', '!=' )
+				->get_results();
```

Comment:
This chunk of code is repeated 4 times.

What do you think of having a function like

```
function get_query_results( $query ) {
     return $query->set_client( $this->client, $this->options->get_ads_id() )
				->where( 'campaign.status', 'REMOVED', '!=' )
				->get_results();
}


$campaign_criterion_results = get_query_results( new AdsCampaignCriterionQuery() );
$campaign_results =  get_query_results( new AdsCampaignQuery() );
```

Reply:
Good point, updated in f4a4342, thanks!